code/game/objects/structures/razorwire.dm

```diff
@@ -74,39 +80,40 @@
 	return razorwire_untangle(entangled)
 
 
-/obj/structure/razorwire/proc/razorwire_untangle(mob/living/M)
-	var/armor_block = null
+/obj/structure/razorwire/proc/razorwire_untangle(mob/living/entangled)
+	entangled.next_move_slowdown += RAZORWIRE_SLOWDOWN //big slowdown
+	do_razorwire_untangle(entangled)
+	visible_message("<span class='danger'>[entangled] disentangles from [src]!</span>")
+	playsound(src, 'sound/effects/barbed_wire_movement.ogg', 25, TRUE)
 	var/def_zone = ran_zone()
-	armor_block = M.run_armor_check(def_zone, "melee")
-	visible_message("<span class='danger'>[M] disentangles from [src]!</span>")
-	playsound(src, 'sound/effects/barbed_wire_movement.ogg', 25, 1)
-	entangled_list -= M
-	M.entangled_by = null
-	M.set_frozen(FALSE)
-	M.apply_damage(rand(RAZORWIRE_BASE_DAMAGE * 0.8, RAZORWIRE_BASE_DAMAGE * 1.2), BRUTE, def_zone, armor_block, TRUE) //Apply damage as we tear free
-	UPDATEHEALTH(M)
-	M.next_move_slowdown += RAZORWIRE_SLOWDOWN //big slowdown
-	DISABLE_BITFIELD(M.restrained_flags, RESTRAINED_RAZORWIRE)
-	UnregisterSignal(M, COMSIG_LIVING_DO_RESIST)
+	var/armor_block = entangled.run_armor_check(def_zone, "melee")
+	entangled.apply_damage(rand(RAZORWIRE_BASE_DAMAGE * 0.8, RAZORWIRE_BASE_DAMAGE * 1.2), BRUTE, def_zone, armor_block, TRUE) //Apply damage as we tear free
+	UPDATEHEALTH(entangled)
 	return TRUE
 
 
+/obj/structure/razorwire/proc/do_razorwire_untangle(mob/living/entangled)
+	UnregisterSignal(entangled, list(COMSIG_PARENT_QDELETING, COMSIG_LIVING_DO_RESIST))
+	LAZYREMOVE(entangled_list, entangled)
+	entangled.entangled_by = null
```

Comment:
uh why not just make entangled from itself in its destroy?

Reply:
actually you can probably just get rid of `/mob/living/var/entangled_by`

Reply:
```
/atom/movable/Uncross(atom/movable/AM, atom/newloc)
	. = ..()
	if(SEND_SIGNAL(src, COMSIG_MOVABLE_UNCROSS, AM) & COMPONENT_MOVABLE_BLOCK_UNCROSS)
```
register on that signal instead on things you entangle

Reply:
Sure. This is Sur code, I didn't stop to improve it because I was fixing other stuff.
I'll see about cleaning it.

Reply:
its basically the cause of the runtime seemed better to treat the cause.